models/bert/bert_tokenizer.hpp

```diff
@@ -0,0 +1,135 @@
+/**
+ * @file models/bert/bert_tokenizer.hpp
+ * @author Mrityunjay Tripathi
+ *
+ * Definition of the BERT Tokenizer.
+ *
+ * @code
+ * @article{Wolf2019HuggingFacesTS,
+ * title = {HuggingFace's Transformers: State-of-the-art Natural Language
+ *          Processing},
+ * author = {Thomas Wolf and Lysandre Debut and Victor Sanh and Julien Chaumond
+ *           and Clement Delangue and Anthony Moi and Pierric Cistac and
+ *           Tim Rault and R'emi Louf and Morgan Funtowicz and Jamie Brew},
+ * journal = {ArXiv},
+ * year = {2019},
+ * volume = {abs/1910.03771}
+ * }
+ * @endcode
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MODELS_BERT_BERT_TOKENIZER_HPP
+#define MODELS_BERT_BERT_TOKENIZER_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+  typename InputDataType = arma::mat,
+  typename OutputDataType = arma::mat
+>
+class BertTokenizer
```

Comment:
If I understand correctly, it's not a network layer. It's just a preprocessor. In this case you don't need the template arguments.